pages/best-practices/best-practices.mdx

```diff
@@ -0,0 +1,293 @@
+Keep it in your codebase, test it before deployment, and write it to your SpiceDB cluster as a part of your continuous integration process.
+This ensures that updates to your schema are properly controlled.
+
+### Load Test
+
+Tags: **operations**
+
+To evaluate the performance and capabilities of your SpiceDB cluster and its underlying datastore, AuthZed provides [Thumper](https://github.com/authzed/thumper) — a load testing tool.
+You can use Thumper to simulate workloads and validate schema updates before deploying them to a production environment.
+
+### Use ZedTokens and “At Least As Fresh” for Best Caching
+
+Tags: **application**
+
+SpiceDB’s fully consistent mode (`fully_consistent`) forces the use of the most recent datastore revision, which might not be the most optimal, and reduces cache hit rate, increasing latency and load on the datastore.
+
+If possible, we recommend using `at_least_as_fresh` with `ZedTokens` instead.
+Capture the `ZedToken` returned by your initial request, then include it in all subsequent calls.
+SpiceDB will guarantee you see a state at least as fresh as that token while still leveraging in-memory and datastore caches to deliver low-latency responses
+
+### Prefer Checking Permissions Instead of Relationships
+
+Tags: **application**
+
+It's possible to make a `Check` call with a relation as the permission.
+Even in a simple schema, we recommend instead that you have a permission that points at the relation and to check the relation.
+This is because if the logic of the check needs to change, it's easy to change the definition of a permission and difficult to change the definition of a relation (it often requires a data migration).
+
+### Enable schema watch cache
+
+Tags: **operations**
+
+In order to minimize load on the database, you can enable schema watch cache using the flag `--enable-experimental-watchable-schema-cache`.
+The schema watch cache is a mechanism that improves performance and responsiveness by caching the currently loaded schema and watching for changes in real time.
+
+While we recommend enabling this, it isn't enabled by default because it requires additional configuration and knowledge of your datastore.
+For Postgres, [`track_commit_timestamp`](https://www.postgresql.org/docs/current/runtime-config-replication.html#GUC-TRACK-COMMIT-TIMESTAMP) must be set to `on` for the Watch API to be enabled.
+For Spanner, there are a maximum of 5 changefeeds available globally for a table, and this consumes one of them.
+
+### Use the Operator
+
+Tags: **operations**
+
+To ensure seamless rollouts, upgrades, and schema migrations, it is recommended to use the SpiceDB Kubernetes Operator if you’re using Kubernetes.
+The Operator automates many operational tasks and helps maintain consistency across environments.
+You can find the official documentation for the SpiceDB operator [here](https://authzed.com/docs/spicedb/concepts/operator).
+
+### Ensure that SpiceDB Can Talk To Itself
```

Comment:
Isn't this critical for performance? I would move this to "strongly recommended"

Reply:
It depends a lot on both the workload and on the constraints of the user's runtime environment. If a user is running in ECS, there isn't a good way to do this. We decided it wasn't broadly applicable enough.